packages/yoastseo/spec/languageProcessing/researches/keywordCountSpec.js

```diff
@@ -22,166 +22,222 @@ const buildMorphologyMockResearcher = function( keyphraseForms ) {
+		expectedCount: 1,
+		expectedMarkings: [ new Mark( { marked: "A string with a <yoastmark class='yoast-text-mark'>$keyword</yoastmark>.",
+			original: "A string with a $keyword." } ) ],
+	},
+	{
+		description: "doesn't count 'key-word' in 'key word'.",
+		paper: new Paper( "A string with a key word.", { keyword: "key-word" } ),
+		keyphraseForms: [ [ "key-word", "key-words" ] ],
+		expectedCount: 0,
+		expectedMarkings: [],
+	},
+	{
+		description: "doesn't count keyphrase instances inside elements we want to exclude from the analysis",
+		paper: new Paper( "There is no <code>keyword</code> in this sentence." ),
+		keyphraseForms: [ [ "keyword", "keywords" ] ],
+		expectedCount: 0,
+		expectedMarkings: [],
+	},
+	{
+		description: "only counts full key phrases (when all keywords are in the sentence once, twice etc.) as matches.",
+		paper: new Paper( "A string with three keys (key and another key) and one word." ),
+		keyphraseForms: [ [ "key", "keys" ], [ "word", "words" ] ],
+		expectedCount: 1,
+		expectedMarkings: [
 			new Mark( { marked: "A string with three <yoastmark class='yoast-text-mark'>keys</yoastmark> (<yoastmark class='yoast-text-mark'>" +
 					"key</yoastmark> and another <yoastmark class='yoast-text-mark'>key</yoastmark>) and one <yoastmark " +
 					"class='yoast-text-mark'>word</yoastmark>.",
-			original: "A string with three keys (key and another key) and one word." } ) ]
-		);
-	} );
+			original: "A string with three keys (key and another key) and one word." } ) ],
+	},
+	{
+		description: "doesn't match singular forms in reduplicated plurals in Indonesian",
+		paper: new Paper( "Lorem ipsum dolor sit amet, consectetur keyword-keyword, keyword adipiscing elit.",
+			{ locale: "id_ID", keyword: "keyword" } ),
+		keyphraseForms: [ [ "keyword", "keywords" ] ],
+		expectedCount: 1,
+		expectedMarkings: [
+			new Mark( {
+				// eslint-disable-next-line max-len
+				marked: "Lorem ipsum dolor sit amet, consectetur <yoastmark class='yoast-text-mark'>keyword</yoastmark>-<yoastmark class='yoast-text-mark'>keyword</yoastmark>, <yoastmark class='yoast-text-mark'>keyword</yoastmark> adipiscing elit.",
+				original: "Lorem ipsum dolor sit amet, consectetur keyword-keyword, keyword adipiscing elit." } ) ],
+	},
+	{
+		description: "counts a single word keyphrase with exact matching",
+		paper: new Paper( "A string with a keyword.", { keyword: "\"keyword\"" } ),
+		keyphraseForms: [ [ "keyword" ] ],
+		expectedCount: 1,
+		expectedMarkings: [ new Mark( { marked: "A string with a <yoastmark class='yoast-text-mark'>keyword</yoastmark>.",
+			original: "A string with a keyword." } ) ],
+	},
+	{
+		description: "with exact matching, a singular single word keyphrase should not be counted if the focus keyphrase is plural",
+		paper: new Paper( "A string with a keyword.", { keyword: "\"keywords\"" } ),
+		keyphraseForms: [ [ "keywords" ] ],
+		expectedCount: 0,
+		expectedMarkings: [],
+		skip: true,
```

Comment:
I could not write the code for this because I need changes from [HTML Parser - Keyphrase density assessment](https://github.com/Yoast/lingo-other-tasks/issues/122). But I already wrote the tests. So in [HTML Parser - Keyphrase density assessment](https://github.com/Yoast/lingo-other-tasks/issues/122) they need to be unskipped and the code for counting exact matches must be written.